ee/server/service/setup_experience.go

```diff
@@ -269,6 +273,30 @@ func (svc *Service) SetupExperienceNextStep(ctx context.Context, host *fleet.Hos
 				level.Warn(svc.logger).Log("msg", "got an error when attempting to enqueue VPP app install", "err", err, "adam_id", app.VPPAppAdamID)
 				app.Status = fleet.SetupExperienceStatusFailure
 				app.Error = ptr.String(err.Error())
+				// At this point we need to check whether the "cancel if software install fails" setting is active,
+				// in which case we'll cancel the remaining pending items.
+				teamID := host.TeamID
+				requireAllSoftware := false
+				if teamID == nil || *teamID == 0 {
+					ac, err := svc.ds.AppConfig(ctx)
+					if err != nil {
+						return false, ctxerr.Wrap(ctx, err, "getting app config")
+					}
+					requireAllSoftware = ac.MDM.MacOSSetup.RequireAllSoftware
+				} else {
+					team, err := svc.ds.Team(ctx, *teamID)
+					if err != nil {
+						return false, ctxerr.Wrap(ctx, err, "load team")
+					}
+					requireAllSoftware = team.Config.MDM.MacOSSetup.RequireAllSoftware
+				}
```

Comment:
Feels like this could be extracted/reused, and this looks slightly different than the way you did it in `orbit.go` code, which seems concerning.